frame/support/procedural/src/key_prefix.rs

```diff
@@ -0,0 +1,107 @@
+// This file is part of Substrate.
+
+// Copyright (C) 2021 Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use proc_macro2::{Span, TokenStream};
+use quote::{ToTokens, format_ident, quote};
+use std::collections::BTreeSet;
+use syn::{Ident, Result};
+
+const ALL_IDENTS: [&'static str; 18] =
```

Comment:
Please generate this. 

You can just generate ident by attaching the number. No need to write down the alphabet.

Reply:
`0` or any other numeric literal is not a proper Rust identifier, sadly. But I believe there is one more way of simplifying this.